packages/components/src/mobile/bottom-sheet/index.native.js

```diff
@@ -89,16 +96,68 @@ class BottomSheet extends Component {
 		Dimensions.addEventListener( 'change', this.onDimensionsChange );
 	}
 
-	keyboardWillShow( e ) {
+	keyboardShow( e ) {
+		if ( ! this.props.isVisible ) {
+			return;
+		}
```

Comment:
I decided to prevent executing any logic related to keyboard show/hide events for non-visible bottom-sheets. Not sure if there would be a case that would require this but at least we should prevent triggering layout animations because they're global.